From a8aefd086bc7c37e431eee799dc6bb24a1bffd7a Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 09 Oct 2022 15:23:41 +0300
Subject: [PATCH] [SW-0] rsa: move buffer allocation inside function

Currently signature verification process needs buffer
size of signature length which is allocated globally
to reduce stack usage. The signature verification functions
are exported by ROM which can be used by other FW components.
There are stalls observed in gaudi2 when the memcmp compares
two buffers one with internal DCCM address and other with DMI DCCM
address when called from ppboot. To avoid such issues, this
patch moves buffer allocation from global to function which
will be allocated from stack and all the buffer allocations
are done on stack.

Change-Id: I37caa73f7083670e2df75d18c6ff7ba084d0e4c7
---

diff --git a/common/rsa/rsa-verify.c b/common/rsa/rsa-verify.c
index 0ad3468..025f681 100644
--- a/common/rsa/rsa-verify.c
+++ b/common/rsa/rsa-verify.c
@@ -9,12 +9,6 @@
 #include <string.h>
 #include <arcboot.h>
 
-/*
- * This array is declared as global, to minimise
- * the usage of stack
- */
-static uint8_t buf_verify[RSA3072_BYTES];
-
 /**
  * Based on Original function - rsa_verify_padding()
  * from Uboot in the same file
@@ -88,6 +82,7 @@
 {
 	int ret;
 	int hash_len;
+	uint8_t buf_verify[sig_len];
 
 	if (!prop || !sig || !hash)
 		return -EIO;
